packages/core/src/directive/directive-builder.ts

```diff
@@ -2,11 +2,13 @@ import { Transformer } from '../utils';
 import { LogicalOperatorBuilder, OperatorBuilder } from '../operator';
 import { DirectiveArgs, Directive } from './directive';
 import { extractRefs } from '../ref';
+import { RecurseBuilder } from './recurse'
 
 export interface DirectiveBuilderArgs {
   filter: LogicalOperatorBuilder | OperatorBuilder;
   cascade: undefined;
   ignoreReflex: undefined;
+  recurse: RecurseBuilder | undefined;
```

Comment:
Don't understand this one - the signature defines the args as plain values only: `recurse(loop?: boolean, depth?: number)`  (though this will be changed to accept an object as discussed in the other comments)

Reply:
Regarding having another class, the reason why I created the Recurse class is because the Directive class expects args to either be an array of objects that have .params method or an object which has .params method. I have no idea why. As a side note, the type system doesn't enforce this because the nowhere is there a type which enforces this expectation.
https://github.com/snipebin/dgraphium/blob/e14cc0440fc7f52e1d7dcf1d1ce59530d00858d8/packages/core/src/directive/directive.ts#L25

Reply:
Good point! But that should be modified to hande case when args is key-value object.